packages/@aws-cdk/cdk/lib/duration.ts

```diff
@@ -169,26 +173,24 @@ class TimeUnit {
   public static readonly Hours = new TimeUnit('hours', 3_600);
   public static readonly Days = new TimeUnit('days', 86_400);
 
-  public readonly inMinutes: number;
-  public readonly inDays: number;
-
   private constructor(public readonly label: string, public readonly inSeconds: number) {
-    this.inMinutes = inSeconds / 60;
-    this.inDays = inSeconds / 86_400;
   }
 
   public toString() {
     return this.label;
   }
 }
 
-function _ensureIntegral(amount: number, multiplier: number, { integral = true }: TimeConversionOptions): number {
+function convert(amount: number, fromUnit: TimeUnit, toUnit: TimeUnit, { integral = true }: TimeConversionOptions) {
+  if (fromUnit.inSeconds === toUnit.inSeconds) { return amount; }
+  const multiplier = fromUnit.inSeconds / toUnit.inSeconds;
```

Comment:
Would compute this *after* the `isUnresolved` check.

Also, I reckon it's preferable to do all multiplications, then the division, in terms of reducing the rounding errors could occur. So I would really say `const value = amount * fromUnit.inSeconds / toUnit.inSeconds;`

Reply:
> Would compute this _after_ the `isUnresolved` check.

There was actually a test that confirmed that even though the value was unresolved, it could still be used (as long as no conversion took place).